common/djangoapps/nodebb/tasks.py

```diff
@@ -0,0 +1,41 @@
+import logging
+
+from celery.exceptions import MaxRetriesExceededError
+from celery.task import task  # pylint: disable=no-name-in-module, import-error
+
+from common.lib.nodebb_client.client import NodeBBClient
+from django.conf import settings
+
+log = logging.getLogger('edx.celery.task')
+
+
+@task(bind=True, default_retry_delay=settings.NODEBB_TASK_DELAY, max_retries=settings.NODEBB_TASK_MAX_TRIES)
+def create_category_on_nodebb_task(self, course_name):
+    status_code, response_body = NodeBBClient().categories.create(name=course_name)
+
+    if status_code != 200:
+        try:
+            self.retry()
+        except MaxRetriesExceededError:
+            log.error(
+                "Error: Can't create nodebb community for the given course {} due to {}".format(
+                    course_name, response_body
+                )
+            )
+    else:
+        log.info('Success: Community created for course {}'.format(course_name))
+
+
+@task(bind=True, default_retry_delay=settings.NODEBB_TASK_DELAY, max_retries=settings.NODEBB_TASK_MAX_TRIES)
+def join_group_on_nodebb_task(self, username, course_name):
+    status_code, response_body = NodeBBClient().users.join(group_name=course_name, user_name=username)
+
+    if status_code != 200:
+        try:
+            self.retry()
```

Comment:
Why we are retrying task in case of non 200 status code? It is quite possible we would have non 200 status code for example in case of resource creation 201. I think we should carefully decide when to retry.

Reply:
I have asked front end dev to give **200** in only one case [when category is successfully created] else provide codes in range of **400's** .

Reply:
what will be the status code if category you are trying to create already exists?